src/Stateless/Graph/UmlDotGraphStyle.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stateless.Graph
+{
+    /// <summary>
+    /// Generate DOT graphs in basic UML style
+    /// </summary>
+    public class UmlDotGraphStyle : IGraphStyle
+    {
+        /// <summary>Get the text that starts a new graph</summary>
+        /// <returns></returns>
+        override internal string GetPrefix()
+        {
+            return "digraph {\n"
+                      + "compound=true;\n"
+                      + "node [shape=Mrecord]\n"
+                      + "rankdir=\"LR\"\n";
+        }
+
+        internal override string FormatOneCluster(SuperState stateInfo)
+        {
+            string stateRepresentationString = "";
+            var sourceName = stateInfo.StateName;
+
+            string label = sourceName;
+
+            if ((stateInfo.EntryActions.Count > 0) || (stateInfo.ExitActions.Count > 0))
+            {
+                label += "\\n----------";
+                foreach (var entryAction in stateInfo.EntryActions)
+                    label += "\\nentry / " + entryAction;
```

Comment:
Maybe use a StringBuilder to concat the strings? Each concat operation creates a new object, and consumes memory. StringBuilder is generally faster.